ESBMC fails to build on LLVM 16

The C++ frontend is preventing the build on LLVM 16 with:

```
src/clang-cpp-frontend/clang_cpp_convert.cpp:565:32: error: no member named 'hasValue' in 'std::optional<const clang::Expr *>'
      assert(ne.getArraySize().hasValue());
             ~~~~~~~~~~~~~~~~~ ^
src/clang-cpp-frontend/clang_cpp_convert.cpp:567:39: error: no member named 'getValue' in 'std::optional<const clang::Expr *>'
      if(get_expr(*(ne.getArraySize().getValue()), size))
```

We should probably change this as it is [deprecated](https://discourse.llvm.org/t/deprecating-llvm-optional-x-hasvalue-getvalue-getvalueor/63716/24)

@kunjsong01 would you mind taking a look?

--- a/src/clang-cpp-frontend/clang_cpp_convert.cpp
+++ b/src/clang-cpp-frontend/clang_cpp_convert.cpp
@@ -563,9 +563,9 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
       new_expr = side_effect_exprt("cpp_new[]", t);
 
       // TODO: Implement support when the array size is empty
-      assert(ne.getArraySize().hasValue());
+      assert(ne.getArraySize());
       exprt size;
-      if(get_expr(*(ne.getArraySize().getValue()), size))
+      if(get_expr(**ne.getArraySize(), size))
         return true;
 
       new_expr.size(size);

